src/ast/mod.rs

```diff
@@ -1216,6 +1216,7 @@ impl Dim {
 }
 
 #[derive(PartialEq, Eq, PartialOrd, Hash, Debug, Copy, Clone)]
+// Dimension Component
```

Comment:
does this comment add anything? I think what a Dimensional Component is as unclear as what a `DimCompo` is. I think this should either be an explanation or the comment is not necessary.